src/charm.py

```diff
@@ -200,6 +204,21 @@ def _on_cluster_relation_changed(self, event: EventBase) -> None:
 
         self._set_status(Status.ACTIVE)
 
+    def _on_secret_changed(self, event: SecretChangedEvent):
+        """Reconfigure services on a secret changed event."""
+        if not event.secret.label:
+            return
+
+        if not self.state.cluster.relation:
+            return
+
+        if event.secret.label == self.state.cluster.data_interface._generate_secret_label(
+            PEER,
+            self.state.cluster.relation.id,
+            "extra",  # type:ignore noqa  -- Changes with the https://github.com/canonical/data-platform-libs/issues/124
+        ):
+            self._on_cluster_relation_changed(event)
```

Comment:
**question** Is this VM specific? Or, could this be abstracted away from the charm.py where we would generally place what is "substrate" and "product" specific? If it is substrate independent, probably somewhere in managers would be a better place, to make the copy/paste as easy as possible

Reply:
Sorry do you mean `secret-changed`?
Or the condition above?

Either way, `secret-changed` is pretty much the same as `relation-changed`, so I guess those two should be kept together.

Reply:
Nope I just meant that if this is substrate independent (meaning this logic can apply as is to both VM and K8s), a better place could be either managers or even handlers (depending whether there is hook-related code), and not `charm.py`. But it is more a suggestion than anything else.